### ISSUE TITLE
Your browser does not support indexedDB :( - or you are in private mode

### ISSUE THREAD

**Issue by bgies** [esphen/RESTED#55]
Have been using Rested for months.... suddenly I'm seeing the error in the subject.... as far as I know I haven't changed any settings. I googled for possible causes, checked that indexedDB is enabled, and that I'm not in privacy mode. So far, no joy...

My FireFox version is 47.0.1.... Rested is set to auto update...

Symptoms... the above error, and my previously save requests don't show up in the left sidebar (just the nice picture of the fox).... if I create a request and try to save it, it tells me that I already have that request, but if I change it slightly so it's not the same as a previously saved request, it doesn't save it. 

using about:config in Firefox
dom.indexedDB.enabled true

Anyone have any suggestions on what I can do to convince Rested that IndexedDB is enabled, and that I'm not in privacy mode? Any other Firefox settings that are worth trying?

**Comment by esphen** [comment]
Hi, thanks for stopping by.

Well, the check if your browser supports IndexedDB, which is run then RESTED starts, is failing. I'll try to reproduce this with your version of FF when I have time, which is probably this weekend. I am not able to reproduce in FF 49, which is what I'm using.

RESTED has not gotten any updates for 17 days, so that this would happen now is weird. Could be an issue with your browser settings. Have you tried with a clean profile?

Another thing you could do is press F12 (open developer tools) and look at the console. If there is anything interesting there, please post it here ("junk after document element" is normal)

**Comment by esphen** [comment]
I have tested in Firefox 47.0.1, and I am not seeing this issue. Did you try with a clean profile? You could also try disabling all other extensions and see if that makes it work again

**Comment by bgies** [comment]
Sorry, didn't see that you had replied... thanks for taking the time to look at this.

I do get this error in the Browser Console: IndexedDB UnknownErr: ActorsParent.cpp:595, but there are no details

Updating to FireFox 48.0 didn't help

**Comment by bgies** [comment]
FYI... updating to FireFox 48.0 did change the error line number... IndexedDB UnknownErr: ActorsParent.cpp:584, although I assume it's the same error :)

**Comment by esphen** [comment]
Yeah, that error is from FIrefox itself, not related. 

Please try the following things:
1. In the console (F12) in RESTED, type `indexedDB.deleteDatabase('RESTED')`
2. Click: tools -> Add-ons -> Extensions, then disable everything but RESTED and restart Firefox. See if you still get the issue
3. Go to Preferences -> Privacy. Ensure you have not set "Never remember history" or "Always use private browsing mode" is off if you use custom settings

**Comment by bgies** [comment]
FYI... updating to FireFox 48.0 did change the error line number... IndexedDB UnknownErr: ActorsParent.cpp:584, although I assume it's the same error :)

**Comment by bgies** [comment]
Well... finally decided to run the indexedDB.deleteDatabase('RESTED') in the console..... unfortunately, it didn't help either...

**Comment by esphen** [comment]
Okay, so that means the issue has nothing to do with the database being corrupted. Did you try the other two things I suggested? I suspect it is a configuration issue

**Comment by Gert-dev** [comment]
I wanted to pitch in here to help you debug, I'm experiencing the same issue. I've tried creating a completely clean Firefox profile and then installing this add-on from there as first add-on, but the same problem ensues.

I also tried opening a console and doing the same steps as this add-on does in a separate tab:

```js
var indexedDB = window.indexedDB || window.mozIndexedDB;
var test = indexedDB.open('_RESTED_TEST_IDB_SUPPORT', 1);
```

`window.indexedDB` exists and `test` is then a `IDBOpenDBRequest { onblocked: null, onupgradeneeded: null, result: IDBDatabase, error: null, source: null, transaction: null, readyState: "done", onsuccess: null, onerror: null }`, so it isn't due to IndexedDB being somehow disabled.

**Comment by esphen** [comment]
Thanks @Gert-dev, that mostly rules out a configuration issue.

Honestly, `IndexedDB` has been nothing but a pain. I'm going to be picking up the `webextensions` branch again soon where `IndexedDB` is replaced by `simple-storage`. Once I finish the porting the problem should go away.

**Comment by Gert-dev** [comment]
Great to hear that you released 1.0.0. I have some... rather unpleasant news to tell you. I've just updated to Firefox 49 (my extensions haven't yet updated to 1.0.0 of this add-on) and now the error has disappeared and my collection is available again.

I assume it was some kind of Firefox bug since it randomly appeared in one version and disappeared again in the next. Still, it gave me a chance to properly back up my requests first.

**Comment by esphen** [comment]
Thanks for the feedback. It's quite possible that is was a Firefox bug, wouldn't be the first time. 

If you are able to export your collections, please do so, as the upgrade to 1.0 will clear your saved data because of the upgrade of the extension to the webextension API. After upgrading you can import your old data.

Anyway, I'm closing this issue because RESTED is no longer using IndexedDB
